packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/gateway/IL1TokenGateway.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+pragma solidity >=0.6.9 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+interface IL1TokenGateway {
```

Comment:
not sure about the naming of this interface, maybe we also want permit support on L2 (for custom token that somehow don't have bridgeburn implemented) so it can be `ITokenGatewayPermit`?

Also we can signal interface support here
https://github.com/offchainlabs/arbitrum/blob/b54f76b9305b02acbd5d4066b1685ba5c7fe9e29/packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol#L368-L374